gammapy/cube/psf_map.py

```diff
@@ -135,12 +157,25 @@ def geom(self):
     @classmethod
     def read(cls, filename, **kwargs):
         """Read a psf_map from file and create a PSFMap object"""
-        psfmap = Map.read(filename, **kwargs)
-        return cls(psfmap)
-
-    def write(self, *args, **kwargs):
+        with fits.open(filename, memmap=False) as hdulist:
+            psfmap = Map.from_hdulist(hdulist, "PSFMAP", "BANDSPSF", "auto")
+            try:
+                expmap = Map.from_hdulist(hdulist, "EXPMAP", "BANDSEXP", "auto")
+            except:
+                expmap = None
+        return cls(psfmap, expmap)
+
+    def write(self, filename, overwrite=False):
         """Write the Map object containing the PSF Library map."""
-        self._psf_map.write(*args, **kwargs)
+        hdulist = self._psf_map.to_hdulist(hdu="PSFMAP", hdu_bands="BANDSPSF")
```

Comment:
Suggest to access puplic properties via the public name even in internal methods. I.e. use `self.psf_map` and `self.exposure_map` here.